languageServer/src/languageHelpers/yamlHelper.ts

```diff
@@ -0,0 +1,220 @@
+import {
+  DocumentSymbol,
+  SymbolKind,
+  Range,
+  Location
+} from 'vscode-languageserver/node'
+import {
+  Document,
+  isNode,
+  isPair,
+  isScalar,
+  Pair,
+  parseDocument,
+  Scalar,
+  Node,
+  visit
+} from 'yaml'
+import { BaseLanguageHelper } from './baseLanguageHelper'
+import * as RegExes from './regexes'
+
+export class YamlHelper extends BaseLanguageHelper<Document> {
+  protected rootNode!: Document
+
+  protected findEnclosingSymbols(offset: number): DocumentSymbol[] {
+    const symbolsFound: Array<DocumentSymbol> = []
+
+    visit(this.rootNode, (_, node) => {
+      if (isNode(node) && node.range) {
+        const range = node.range
+        const nodeStart = range[0]
+        const nodeEnd = range[2]
+        const nodeContainsTheOffset = offset >= nodeStart && offset <= nodeEnd
+
+        if (nodeContainsTheOffset) {
+          symbolsFound.push(...this.yamlNodeToDocumentSymbols(node, range))
+        }
+      }
+    })
+
+    return symbolsFound
+  }
+
+  protected parse(source: string) {
+    return parseDocument(source)
+  }
+
+  protected getPropertyLocation(
+    pathArray: Array<string | number>
+  ): Location | null {
+    const node = this.rootNode.getIn(pathArray, true)
+
+    if (isNode(node) && node.range) {
+      const [nodeStart, , nodeEnd] = node.range
+      const start = this.positionAt(nodeStart)
+      const end = this.positionAt(nodeEnd)
+      const range = Range.create(start, end)
+      return Location.create(this.textDocument.uri, range)
+    }
+
+    return null
+  }
+
+  protected toJSON(): unknown {
+    return this.rootNode.toJS()
```

Comment:
Why `toJS()` instead of `toJSON()`? Might be worth a comment.

Reply:
Makes more sense, thanks!